specification/translation/Azure.AI.TextTranslation/models-languages.tsp

```diff
@@ -112,28 +114,42 @@ model CommonScriptModel {
   nativeName: string;
 
   @doc("Directionality, which is rtl for right-to-left languages or ltr for left-to-right languages.")
-  dir: string;
+  dir: LanguageDirectionality;
 }
 
-alias CommonDictionaryLanguageModel = {
-  ...CommonLanguageModel;
-
-  @doc("Directionality, which is rtl for right-to-left languages or ltr for left-to-right languages.")
-  dir: string;
-};
-
 @doc("Properties ot the source dictionary language")
 model SourceDictionaryLanguage {
-  ...CommonDictionaryLanguageModel;
+  ...BaseLanguage;
+
+  @doc("Directionality, which is rtl for right-to-left languages or ltr for left-to-right languages.")
+  dir: LanguageDirectionality;
 
   @doc("List of languages with alterative translations and examples for the query expressed in the source language.")
   translations: TargetDictionaryLanguage[];
 }
 
 @doc("Properties of the target dictionary language")
 model TargetDictionaryLanguage {
-  ...CommonDictionaryLanguageModel;
+  ...BaseLanguage;
+
+  @doc("Directionality, which is rtl for right-to-left languages or ltr for left-to-right languages.")
+  dir: LanguageDirectionality;
 
   @doc("Language code identifying the target language.")
   code: string;
 }
+
+@doc("Language Directionality")
+union LanguageDirectionality {
+  LeftToRight: "ltr",
+  RightToLeft: "rtl",
+  string,
```

Comment:
Do we really need to make this directionality open-ended? Do we expect customers to pass a custom directionality that is not defined here?

Reply:
We should consider the same for the unions `ProfanityAction` and `ProfanityMarker`, really for all Unions, make sure we really want to make the set open ended

Reply:
Done for all fixed enums.